classes/CartRule.php

```diff
@@ -1575,34 +1586,31 @@ public static function getCartsRuleByCode($name, $id_lang, $extended = false)
     /**
      * CartRules compare function to use the Product and the rules
      *
-     * @param array $arrayProduct List Products,
-     * @param array $arrayRules List Rules,
-     * @param string $type type voucher,
+     * @param array $products List of Products from the cart,
+     * @param array $eligibleProducts List of Product eligible for rules,
+     * @param string $ruleType name of the rule,
      *
-     * @return int 0 = same
-     *             1 = different
+     * @return array Product selected who are eligible
      */
-    public function cartRulesCompare($arrayProduct, $arrayRules, $type)
+    protected function filterProducts($products, $eligibleProducts, $ruleType)
```

Comment:
This method added on a PR just before. No problem, this is just to avoid this kind of problem!